### ISSUE TITLE
May I ask why the test results of last.fm data set are about 10% higher than those in the paper

### ISSUE THREAD

**Issue by liwenwei5110** [zzaebok/KGCN-pytorch#5]
May I ask why the test results of last.fm data set are about 10% higher than those in the paper

**Comment by zzyo-yu** [comment]
I also got the same result. Do you know the reason now?

**Comment by liwenwei5110** [comment]
> 

i dont know

**Comment by kaishuibujiatang** [comment]
You can share the metrics code that you wrote？thank you very much。

**Comment by kaishuibujiatang** [comment]
你可以分享你写的实验结果评价标准代码？非常感谢。

**Comment by kaishuibujiatang** [comment]
可以分享一下你写的实验结果评价标准代码？非常感谢。

**Comment by zzyo-yu** [comment]
> 可以分享一下你写的实验结果评价标准代码？非常感谢。

I used his code without any modifications.

**Comment by kaishuibujiatang** [comment]
论文里面的评价指标是AUC和F1，他的代码实验结果评价指标只有 损失值和roc_auc_score，是怎么看出来比论文里面的结果好10%呀？

**Comment by kaishuibujiatang** [comment]
> > 可以分享一下你写的实验结果评价标准代码？非常感谢。
> 
> I used his code without any modifications.
论文里面的评价指标是AUC和F1，他的代码实验结果评价指标只有 损失值和roc_auc_score，是怎么看出来比论文里面的结果好10%呀？

**Comment by kaishuibujiatang** [comment]
> May I ask why the test results of last.fm data set are about 10% higher than those in the paper

可以分享一下你写的实验结果评价标准代码？非常感谢。

**Comment by liwenwei5110** [comment]
>

**Comment by liwenwei5110** [comment]
> 

我也没改，按照源代码跑的结果，具体明天我看一下

**Comment by kaishuibujiatang** [comment]
好的，感谢感谢

**Comment by liwenwei5110** [comment]
roc_auc_score算出来的结果就是Auc的值，我跑出来的结果roc_auc_score=0.89568，F1=0.80918


---Original---
From: ***@***.***&gt;
Date: Tue, Aug 2, 2022 22:19 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;&quot;State ***@***.***&gt;;
Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5)




 
好的，感谢感谢
 
&mdash;

**Comment by kaishuibujiatang** [comment]
> roc_auc_score算出来的结果就是Auc的值，我跑出来的结果roc_auc_score=0.89568，F1=0.80918
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Tue, Aug 2, 2022 22:19 PM To: ***@***.***&gt;; Cc: ***@***.***&gt;;&quot;State ***@***.***&gt;; Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5) 好的，感谢感谢 &mdash; Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: ***@***.***&gt;

嗯嗯，感谢感谢，可以分享一下F1是怎么写的吗

**Comment by liwenwei5110** [comment]
from&nbsp;sklearn.metrics&nbsp;import&nbsp;roc_auc_score,f1_score这两个函数，你跑出来结果怎么样呢，可以告诉我一下吗


---Original---
From: ***@***.***&gt;
Date: Wed, Aug 3, 2022 16:58 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;&quot;State ***@***.***&gt;;
Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5)




  
roc_auc_score算出来的结果就是Auc的值，我跑出来的结果roc_auc_score=0.89568，F1=0.80918
 …
 ---Original--- From: @.&gt; Date: Tue, Aug 2, 2022 22:19 PM To: @.&gt;; Cc: @.&gt;;"State @.&gt;; Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5) 好的，感谢感谢 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: @.***&gt;
  
嗯嗯，感谢感谢，可以分享一下F1是怎么写的吗
 
&mdash;

**Comment by kaishuibujiatang** [comment]
Auc好像是0.86左右，f1没有跑，每天再跑一边，给你具体结果

**Comment by kaishuibujiatang** [comment]
Auc好像是0.86左右，f1没有跑，每天再跑一边，给你具体结果

> from&nbsp;sklearn.metrics&nbsp;import&nbsp;roc_auc_score,f1_score这两个函数，你跑出来结果怎么样呢，可以告诉我一下吗
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Wed, Aug 3, 2022 16:58 PM To: ***@***.***&gt;; Cc: ***@***.***&gt;;&quot;State ***@***.***&gt;; Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5) roc_auc_score算出来的结果就是Auc的值，我跑出来的结果roc_auc_score=0.89568，F1=0.80918 … ---Original--- From: @.&gt; Date: Tue, Aug 2, 2022 22:19 PM To: @.&gt;; Cc: @.&gt;;"State @.&gt;; Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5) 好的，感谢感谢 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: @.***&gt; 嗯嗯，感谢感谢，可以分享一下F1是怎么写的吗 &mdash; Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: ***@***.***&gt;

Auc好像是0.86左右，f1没有跑，每天加上f1再跑一边，告诉你具体结果

**Comment by kaishuibujiatang** [comment]
> from&nbsp;sklearn.metrics&nbsp;import&nbsp;roc_auc_score,f1_score这两个函数，你跑出来结果怎么样呢，可以告诉我一下吗
> […](#)
> ---Original--- From: ***@***.***&gt; Date: Wed, Aug 3, 2022 16:58 PM To: ***@***.***&gt;; Cc: ***@***.***&gt;;&quot;State ***@***.***&gt;; Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5) roc_auc_score算出来的结果就是Auc的值，我跑出来的结果roc_auc_score=0.89568，F1=0.80918 … ---Original--- From: @.&gt; Date: Tue, Aug 2, 2022 22:19 PM To: @.&gt;; Cc: @.&gt;;"State @.&gt;; Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5) 好的，感谢感谢 — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: @.***&gt; 嗯嗯，感谢感谢，可以分享一下F1是怎么写的吗 &mdash; Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you modified the open/close state.Message ID: ***@***.***&gt;

Auc好像是0.86左右，f1没有跑，每天加上f1再跑一边，给你具体结果

**Comment by kaishuibujiatang** [comment]
Auc是0.85610，f1是0.779465

**Comment by liwenwei5110** [comment]
你这个结果也是比论文高挺多的，你知道原因吗


---Original---
From: ***@***.***&gt;
Date: Thu, Aug 4, 2022 10:18 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;&quot;State ***@***.***&gt;;
Subject: Re: [zzaebok/KGCN-pytorch] May I ask why the test results of last.fmdata set are about 10% higher than those in the paper (Issue #5)




 
Auc是0.85610，f1是0.779465
 
&mdash;

**Comment by kaishuibujiatang** [comment]
我也不知道

**Comment by violetingithub** [comment]
请问您能分享一下加入了AUC，F1score的文件或者代码吗？我加入了评价指标的代码，但是跑出来超级低，我不知道哪里出了问题。能否看一下您的代码？非常感谢

**Comment by violetingithub** [comment]
> 请问您能分享一下加入了AUC，F1score的文件或者代码吗？我加入了评价指标的代码，但是跑出来超级低，我不知道哪里出了问题。能否看一下您的代码？非常感谢

暂时不用了，我知道了，我按照论文设置的batchsize跑出来很低auc0.5，按照本复现代码的作者设置的batchsize跑出来很好。

**Comment by DW934** [comment]
outputs_predict = [int(item > 0.5) for item in outputs.cpu().detach().numpy()]
total_f1 += f1_score(labels.cpu().detach().numpy(), outputs_predict)
print('[Epoch {}]test_f1: '.format(epoch + 1), total_f1 / len(test_loader))
这是我加的f1
